src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -16,6 +16,39 @@ public Rectangle(Integer width, Integer length, String color) {
         this.color = color;
     }
 
+    @Override
+    public int hashCode() {
+        int result = 17;
+        result = 31 * result + (width != null ? width.hashCode() : 0);
+        result = 31 * result + (length != null ? length.hashCode() : 0);
+        result = 31 * result + (color != null ? color.hashCode() : 0);
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+
+        if (obj == null || !obj.getClass().equals(Rectangle.class)) {
+            return false;
+        }
+
+        try {
+            Rectangle rectangle = (Rectangle) obj;
+            return (width == null || rectangle.width == null) ? width == rectangle.width
```

Comment:
```suggestion
            return width == null  ? width == rectangle.width
```
that should be enough.